docs-requirements.txt

```diff
@@ -0,0 +1,8 @@
+sphinx~=2.4
+sphinx-rtd-theme~=0.4
+sphinx-autodoc-typehints~=1.10.2
+# External
+opentelemetry-api
```

Comment:
I believe requirements files are used to simply specify the version numbers of the installations. We should keep the actual installations in `deps`.